src/webserver/DOSGuard.h

```diff
@@ -69,6 +73,333 @@ class BasicDOSGuard : public BaseDOSGuard
     clio::Logger log_{"RPC"};
 
 public:
+    /**
+     * @brief Checks if raw IPv4 address is valid
+     *
+     * @param ip Raw IPv4 address to check
+     */
+    [[nodiscard]] bool
+    checkParsedIPv4Address(std::string const& ip) const
+    {
+        try
+        {
+            boost::asio::ip::address_v4::from_string(ip);
+            return true;
+        }
+
+        catch (const std::exception& e)
+        {
+            return false;
+        }
+    }
+
+    /**
+     * @brief Checks if raw IPv6 address is valid
+     *
+     * @param ip Raw IPv6 address to check
+     */
+    [[nodiscard]] bool
+    checkParsedIPv6Address(std::string const& ip) const
+    {
+        try
+        {
+            boost::asio::ip::address_v6::from_string(ip);
+            return true;
+        }
+
+        catch (const std::exception& e)
+        {
+            return false;
+        }
+    }
+
+    /**
+     * @brief Checks if IPv4 address has a valid CIDR notation
+     *
+     * @param ip The IP address (IPv4)
+     */
+    [[nodiscard]] bool
+    checkCIDRIPv4(std::string const& ip) const
```

Comment:
to validate a valid subnet , i think we can call "make_network_v4". then you can hold a subnet or subnet hosts in memory. When you need to check, you don't need to call make_network again. Imagine, when we receive a request, we want to do it timely.